packages/itwinui-react/src/core/SearchBox/SearchBox.tsx

```diff
@@ -0,0 +1,39 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import { InputProps } from '../Input';
+import { useTheme } from '../utils';
+
+export type SearchBoxProps = {
+  /**
+   *
+   */
+  startAdornment?: React.ReactNode | React.ReactNode[];
+  /**
+   *
+   */
+  endAdornment?: React.ReactNode | React.ReactNode[];
+} & InputProps;
+
+/**
+ * // TODO:
+ * Searchbox component.
+ * Used for searches :)
+ * @example
+ * Example usages go here!
+ */
+export const SearchBox = (props: SearchBoxProps) => {
```

Comment:
so why is this component called searchbox? i think it can be used in many cases, not just search

Reply:
Because people want searchbox. This "search" layout could be achieved with no/minimal css changes (input with button) but devs still want separate search component.
Also, we are planning to expand this with functionality, like save 5 last searches.

CSS is not using search in the name, so we could create separate container for this new flex-container css and reuse it in SearchBox component.

Reply:
> Also, we are planning to expand this with functionality, like save 5 last searches.

How will this work? Our components should not contain any application logic, so it sounds a bit out of scope for us.

Reply:
Added animation to searchbox to make it more complex. Created InputFlexContainer for internal use.